### PR TITLE
Fix:slow memory leak may be in kubelet podworkers.isWorking

### LINKED ISSUE #102723: slow memory leak may be in kubelet podworkers.isWorking
#### What happened:

As the p.isWorking won't be cleared while pod was deleted, so while creating or deleting pods repeatedly, length of p.isWorking keep increasing

```
I0610 00:56:22.686625   12730 pod_workers.go:240] p.isWorking length 8
I0610 00:56:27.686615   12730 pod_workers.go:240] p.isWorking length 8

I0610 01:01:41.686553   12730 pod_workers.go:240] p.isWorking length 13
I0610 01:01:42.686687   12730 pod_workers.go:240] p.isWorking length 13
I0610 01:01:48.686530   12730 pod_workers.go:240] p.isWorking length 13


I0610 01:03:00.003440   12730 pod_workers.go:240] p.isWorking length 18


I0610 01:03:55.331554   12730 pod_workers.go:240] p.isWorking length 23
I0610 01:03:55.343556   12730 pod_workers.go:240] p.isWorking length 23
```

#### What you expected to happen:

length of p.isWorking does not keep increasing

#### How to reproduce it (as minimally and precisely as possible):
 
creating or deleting pods repeatedly

#### Environment:
- Kubernetes version (use `kubectl version`): 1.19.4
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`): 
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by 249043822**
#### What type of PR is this?

/kind bug
#### Which issue(s) this PR fixes:

Fixes #102723

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/pod_workers.go [hunk 1]
```diff
@@ -282,7 +282,8 @@ func (p *podWorkers) checkForUpdates(uid types.UID) {
 		p.podUpdates[uid] <- workUpdate
 		delete(p.lastUndeliveredWorkUpdate, uid)
 	} else {
-		p.isWorking[uid] = false
+		// put this line in removeWorker may cause dead lock, so keep reset it here
+		delete(p.isWorking, uid)
```
Comment (liggitt) [inline pkg/kubelet/pod_workers.go:286]: this needs an accompanying test demonstrating the leak and that this fix works properly

**Review by ehashman — COMMENTED** [review]
Please add a test, per @liggitt.

/triage accepted
/priority important-longterm

**Comment by 249043822** [comment]
@liggitt @ehashman test case added

**Comment by matthyx** [comment]
I do agree this should be deleted rather than set to `false` but it's probably more for correctness than real memory usage leak.
/lgtm

**Review by mrunalp — APPROVED** [review]

**Comment by smarterclayton** [comment]
Note that #102344 also fixed this bug, and that change will remove this code path.  Please take a look at these changes to make sure your fix carries over.

**Pull request merged**